shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedProcessorSyncServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.redhat.service.smartevents.shard.operator.v2;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.smartevents.infra.v2.api.models.ConditionStatus;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ConditionDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ProcessorDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ResourceStatusDTO;
+import com.redhat.service.smartevents.shard.operator.v2.converters.ResourceStatusConverter;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedProcessor;
+
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_PROCESSOR_DELETED_NAME;
+
+@ApplicationScoped
+public class ManagedProcessorSyncServiceImpl implements ManagedProcessorSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessorSyncServiceImpl.class);
+
+    @Inject
+    ManagerClient managerClient;
+
+    @Inject
+    ManagedProcessorService managedProcessorService;
+
+    @Override
+    public void syncManagedProcessorWithManager() {
+        managerClient.fetchProcessorsForDataPlane()
+                .onItem()
+                .invoke(this::processDelta)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully processed ManagedProcessor deltas"),
+                        error -> LOGGER.error("Failed to process ManagedProcessor deltas"));
+    }
+
+    @Override
+    public void syncManagedProcessorStatusBackToManager() {
+        managerClient.fetchProcessorsForDataPlane()
+                .onItem()
+                .transform(this::transformToProcessorStatus)
+                .invoke(this::notifyProcessorStatus)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully sync ManagedProcessor status with Manager"),
+                        error -> LOGGER.error("Failed to sync ManagedProcessor status with Manager", error));
+    }
```

Comment:
Might be a good idea to avoid calling `managerClient.fetchProcessorsForDataPlane()` when you want to sync back the status as this would force the manager to fetch all the instances from the DB another time - it is not really necessary

Reply:
We already have a discussion about it and its finalize to keep both processes seperate.

Reply:
Thanks, could you point me to the motivations/discussions or provide a good reason here to do it?